### DISCUSSION TITLE
V 2.0.0!!!!

### DISCUSSION THREAD

**Discussion by nklayman** [nklayman/vue-cli-plugin-electron-builder#1363] [General]
This release has been long overdue and this plugin has been in the RC state for way too long, sorry for the delay. Anyways, better late than never, so here it is! If you were running the latest RC previously, there aren't any breaking changes or required migration steps. Thank you to all my [supporters](https://github.com/nklayman/vue-cli-plugin-electron-builder#supporters) who have funded this project and made this release possible :heart::heart:!!

## Breaking Changes from v1.x
- The `testWithSpectron` function requires you pass the  `spectron` module as it's first arg, allowing you to specify the version of spectron to be used. You will need to install [spectron](https://electronjs.org/spectron) as a `devDependency`
- Jest tests must be run with `testEnvironment` set to `node`, see [the examples](https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/testingAndDebugging.html#testing) for more details
- [Node integration is disabled by default](https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/security.html#node-integration)
- [Electron devtools installer](https://www.npmjs.com/package/electron-devtools-installer) should be used over the `installVueDevtools` function

A few major deps were upgraded:
- teser-webpack-plugin ^1.1.0 -> ^2.0.0
- webpack-chain ^5.0.0 -> ^6.0.0
- execa: ^1.0.0 -> ^4.0.0
- electron-builder: ^21.2.0 -> 22.2.0

These upgrades may cause breaking changes depending on your specific application.

The internal tests are now run on vue cli v4, and it is recommended that you update your app to vue cli v4 as well.

## Fixes and Features
- Support latest electron versions 
- nodeIntegration is now optional (and disabled by default), #610
- electron-builder will now work on macos catalina, #617
- You can now use the proper spectron version, #467
- Removed fonts duplication in build, #694 
- You can now bundle preload files with webpack ([docs](https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/guide.html#preload-files)), #613
- New native dep checker, try it out now! (#861)
- Vue devtools enabled by default on Electron v8/9 (#378)
- Support `package.json` without `dependencies` field (#828)
- Support `electron` being listed in `dependencies` as well as `devDependencies` (#829)
- Support nested pages (#786)
- And a lot more smaller fixes

## Migrating a v1.x Project:

1. Run `(yarn add | npm install) -D vue-cli-plugin-electron-builder@latest` to install the 2.0 version of the plugin.
2. In your `background.(js|ts)`, replace `nodeIntegration:true` with:
```js
// Use pluginOptions.nodeIntegration, leave this alone 
// See nklayman.github.io/vue-cli-plugin-electron-builder/guide/security.html#node-integration for more info 
   nodeIntegration: process.env.ELECTRON_NODE_INTEGRATION 
```
3. If your app requires `nodeIntegration`, enable it in `vue.config.js`:
```js
module.exports = {
  pluginOptions: {
    electronBuilder: {
      nodeIntegration: true
    }
  }
}
```
4. If you used the `testWithSpectron` function, install the proper version of spectron according to the [version map](https://github.com/electron-userland/spectron#version-map), and pass the spectron import at the first argument of all of your `testWithSpectron` calls. See [the docs](https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/testingAndDebugging.html#testing) for more details.
5. If you used jest, set the test environment to node in each electron test file. See [the docs](https://nklayman.github.io/vue-cli-plugin-electron-builder/guide/testingAndDebugging.html#testing) for more details.
6. Update your main process file (`src/background.js` by default) to use electron-devtools-installer:
```diff
import {
  createProtocol,
- installVueDevtools
} from 'vue-cli-plugin-electron-builder/lib'
+ import installExtension, { VUEJS_DEVTOOLS } from 'electron-devtools-installer'

... (around line 77)

-await installVueDevtools()
+await installExtension(VUEJS_DEVTOOLS)
```
And then run `npm i -D electron-devtools-installer`/`yarn add -D electron-devtools-installer`.

After following these steps, your project should be good to go with v2.0!

<hr /><em>This discussion was created from the release <a href='https://github.com/nklayman/vue-cli-plugin-electron-builder/releases/tag/v2.0.0'>V 2.0.0!!!!</a>.</em>

**Comment by aleksey-hoffman** [comment]
Just updated to 2.0.0 and tested both `electron:serve` and `electron:build`. Everything seems to be working fine so far.
Thank you for the release!

**Comment by klren0312** [comment]
Good

**Comment by JenuelDev** [comment]
AWesome, hope that this plugin keeps improving :)

**Comment by JenuelDev** [comment]
Is their an electron plugin for vue 3?

**Reply by nklayman** [reply]
This plugin supports Vue 3

**Reply by JenuelDev** [reply]
does it support VIte? or Vue-CLI only?

**Reply by nklayman** [reply]
Vue CLI only, Vite would require a full rewrite.
